File: crates/cartograph-parser/tests/fixtures/python/symbols.py
"""Fixture: declarations, nesting and __all__."""

BASE = "/api/v1"
_private_const = 3
__all__ = ["fetch_orders", "Cart"]


def plain():
    pass


async def fetch_orders():
    pass


def outer():
    def nested():
        pass
    return nested


class Cart:
    limit = 10

    def add(self, amount):
        self.total += amount

    async def sync(self):
        pass

    @staticmethod
    def helper():
        pass


class _Hidden:
    pass

File: crates/cartograph-parser/tests/fixtures/symbols.ts
// Fixture: symbol declarations and export forms.
const BASE = "/api/v1";
let counter = 0;

export function plain(): void {}

export async function fetchOrders(): Promise<void> {}

function inner_host(): void {
    function nested(): void {}
}

export class Cart {
    total = 0;
    add(amount: number): void {
        this.total += amount;
    }
    async sync(): Promise<void> {}
}

export interface Order {
    id: string;
}

export type OrderId = string;

export enum Status {
    Open,
    Closed,
}

class Hidden {}

export { BASE };
export default Hidden;
